File: apps/mac-shell/Sources/Foundation/Storage/PersistenceBoundary.swift
import Foundation

enum PersistenceScope {
    case threadsCache
    case timelineCache
    case securityAudit

    var fileName: String {
        switch self {
        case .threadsCache:
            return "threads-cache.sqlite"
        case .timelineCache:
            return "timeline-cache.sqlite"
        case .securityAudit:
            return "security-audit.sqlite"
        }
    }

    var storesSensitiveData: Bool {
        false
    }
}

struct PersistenceBoundary {
    let baseDirectory: URL

    var stateDirectory: URL {
        baseDirectory.appendingPathComponent("state", isDirectory: true)
    }

    func sqliteURL(for scope: PersistenceScope) -> URL {
        stateDirectory.appendingPathComponent(scope.fileName, isDirectory: false)
    }

    func requiresSecureStore(for scope: PersistenceScope) -> Bool {
        scope.storesSensitiveData
    }
}

File: apps/mobile/lib/foundation/storage/persistence_boundary.dart
enum PersistenceScope { threadsCache, timelineCache, securityAudit }

extension PersistenceScopeMetadata on PersistenceScope {
  String get fileName {
    switch (this) {
      case PersistenceScope.threadsCache:
        return 'threads-cache.sqlite';
      case PersistenceScope.timelineCache:
        return 'timeline-cache.sqlite';
      case PersistenceScope.securityAudit:
        return 'security-audit.sqlite';
    }
  }

  bool get storesSensitiveData => false;
}

class PersistenceBoundary {
  const PersistenceBoundary({required this.baseDirectory});

  final String baseDirectory;

  String get stateDirectory => '$baseDirectory/state';

  String sqlitePathFor(PersistenceScope scope) =>
      '$stateDirectory/${scope.fileName}';

  bool requiresSecureStore(PersistenceScope scope) => scope.storesSensitiveData;
}
